util/src/main/java/org/cryptomator/util/crypto/CryptoOperationsFactory.java

```diff
@@ -2,21 +2,28 @@
 
 class CryptoOperationsFactory {
 
-	private static volatile CryptoOperations cryptoOperations;
+	private static volatile CryptoOperations cryptoOperationsCBC;
+	private static volatile CryptoOperations cryptoOperationsGCM;
 
-	public static CryptoOperations cryptoOperations() {
-		if (cryptoOperations == null) {
-			synchronized (CryptoOperations.class) {
-				if (cryptoOperations == null) {
-					cryptoOperations = createCryptoOperations();
+	public static CryptoOperations cryptoOperations(CryptoMode mode) {
+		if (mode == CryptoMode.CBC) {
```

Comment:
Again: I'd use a switch-statement here.

Reply:
Fixed in https://github.com/cryptomator/android/commit/9615090966def68d10b86241c37ddace9946c0ef